etna/transforms/timestamp/holiday.py

```diff
@@ -145,15 +148,78 @@ def fit(self, ts: TSDataset) -> "HolidayTransform":
         -------
         :
             The fitted transform instance.
+
+        Raises
+        ------
+        ValueError:
+            if index timestamp is integer and ``in_column`` isn't set
+        ValueError:
+            if external timestamp isn't datetime
+        ValueError
+            if in ``days_count`` mode external timestamp doesn't have frequency
+        ValueError
+            if in ``days_count`` mode external timestamp doesn't have the same frequency for all segments
         """
         if self.in_column is None:
+            if self._mode is HolidayTransformMode.days_count:
+                if ts.freq is None:
+                    raise ValueError("Transform can't work with integer index, parameter in_column should be set!")
+                self._freq = ts.freq
+            else:
+                # set some value that doesn't really matter
+                self._freq = object()  # type: ignore
```

Comment:
We have already set `self._freq` in `__init__`, this is redundant line.

Reply:
This value will be different from `_DEFAULT_FREQ`, because it is a different object.